librz/core/cmd/cmd_analysis.c

```diff
@@ -2742,6 +2673,166 @@ static bool cmd_aea(RzCore *core, int mode, ut64 addr, int length) {
+			break;
+		}
+		rz_core_reg_update_flags(core);
+		ut64 addr = rz_num_get(core->num, pc);
+		op = rz_core_analysis_op(core, addr, RZ_ANALYSIS_OP_MASK_BASIC | RZ_ANALYSIS_OP_MASK_HINT);
+		if (!op) {
+			break;
+		}
+		if (op->type == RZ_ANALYSIS_OP_TYPE_SWI) {
+			RZ_LOG_ERROR("syscall at 0x%08" PFMT64x "\n", addr);
+			break;
+		} else if (op->type == RZ_ANALYSIS_OP_TYPE_TRAP) {
+			RZ_LOG_ERROR("trap at 0x%08" PFMT64x "\n", addr);
+			break;
+		}
+		rz_analysis_op_free(op);
+		op = NULL;
+		if (core->analysis->esil->trap || core->analysis->esil->trap_code) {
+			break;
+		}
+	}
+	if (op) {
+		rz_analysis_op_free(op);
+		op = NULL;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// aecc
+RZ_IPI RzCmdStatus rz_analysis_continue_until_call_handler(RzCore *core, int argc, const char **argv) {
+	const char *pc = rz_reg_get_name(core->analysis->reg, RZ_REG_NAME_PC);
+	RzAnalysisOp *op;
+	for (;;) {
+		if (!rz_core_esil_step(core, UT64_MAX, NULL, NULL, false)) {
+			break;
+		}
+		rz_core_reg_update_flags(core);
+		ut64 addr = rz_num_get(core->num, pc);
+		op = rz_core_analysis_op(core, addr, RZ_ANALYSIS_OP_MASK_BASIC);
+		if (!op) {
+			break;
+		}
+		if (op->type == RZ_ANALYSIS_OP_TYPE_CALL || op->type == RZ_ANALYSIS_OP_TYPE_UCALL) {
+			eprintf("call at 0x%08" PFMT64x "\n", addr);
+			break;
+		}
+		rz_analysis_op_free(op);
+		op = NULL;
+		if (core->analysis->esil->trap || core->analysis->esil->trap_code) {
+			break;
+		}
+	}
+	if (op) {
+		rz_analysis_op_free(op);
+		op = NULL;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// aecu
+RZ_IPI RzCmdStatus rz_analysis_continue_until_addr_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_esil_step(core, rz_num_math(core->num, argv[1]), NULL, NULL, false);
+	rz_core_reg_update_flags(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aecue
+RZ_IPI RzCmdStatus rz_analysis_continue_until_esil_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_esil_step(core, UT64_MAX, argv[1], NULL, false);
+	rz_core_reg_update_flags(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aei
+RZ_IPI RzCmdStatus rz_analysis_esil_init_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_analysis_esil_reinit(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aei-
+RZ_IPI RzCmdStatus rz_analysis_esil_deinit_handler(RzCore *core, int argc, const char **argv) {
+	RzAnalysisEsil *esil = core->analysis->esil;
+	if (esil) {
+		sdb_reset(esil->stats);
+	}
+	rz_analysis_esil_free(esil);
+	core->analysis->esil = NULL;
+	return RZ_CMD_STATUS_OK;
+}
+
+// aeip
+RZ_IPI RzCmdStatus rz_analysis_esil_init_p_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_analysis_set_reg(core, "PC", core->offset);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aeim
+RZ_IPI RzCmdStatus rz_analysis_esil_init_mem_handler(RzCore *core, int argc, const char **argv) {
+	ut64 addr = argc > 0 ? rz_num_math(core->num, argv[0]) : UT64_MAX;
+	ut32 size = argc > 1 ? (ut32)rz_num_math(core->num, argv[1]) : UT32_MAX;
```

Comment:
thre is no reason to check the number of args, since this is done by the parser, just be sure to set the correct amount on the yaml. if instead you need optional, then yes, you do need to check for them, but argc will be always > 0.
also in case of optional args, these needs to be defined on the yaml.

Reply:
OK

Reply:
just to be clear, these args are behaving like the one in the main. so argv[0] is the command name